File: cf_java/prob_2057B.java
import java.util.*;

public class prob_2057B {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        while(t-- > 0) {
            int n = sc.nextInt();
            int k = sc.nextInt();
            int[] arr = new int[n];
            HashMap<Integer, Integer> map = new HashMap<>();
            for(int i = 0; i < n; i++) {
                arr[i] = sc.nextInt();
                map.put(arr[i], map.getOrDefault(arr[i], 0) + 1);
            }
            List<Integer> freqList = new ArrayList<>(map.values());
            Collections.sort(freqList);
            int uniqueCount = freqList.size();
            for(int freq : freqList) {
                if(k >= freq) {
                    k -= freq;
                    uniqueCount--;
                } else {
                    break;
                }
            }
            if (uniqueCount == 0) {
                uniqueCount = 1;
            }
            System.out.println(uniqueCount);
        }
        sc.close();
    }
}

File: cf_cpp/probs/prob_2057B.cpp
#include <bits/stdc++.h>
using namespace std;

void solve_prob_2057B() {
    int n; cin >> n;
    int k; cin >> k;
    vector<int> arr(n);
    unordered_map<int, int> frequencyMap;
    for (int i = 0; i < n; i++) {
        cin >> arr[i];
        frequencyMap[arr[i]]++;
    }
    vector<int> freqList;
    for (const auto& pair : frequencyMap) {
        freqList.push_back(pair.second);
    }
    sort(freqList.begin(), freqList.end());
    int uniqueCount = freqList.size();
    for (int freq : freqList) {
        if (k >= freq) {
            k -= freq;
            uniqueCount--;
        } else {
            break;
        }
    }
    if (uniqueCount == 0) {
        uniqueCount = 1;
    }

    cout << uniqueCount << endl;
}

void prob_2057B() {
    int t = 1;
    cin >> t;
    while (t--) {
        solve_prob_2057B();
    }
}